core/hosting/HostWithCoreClrHost/src/SampleHost.cpp

```diff
@@ -0,0 +1,324 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+
+// https://github.com/dotnet/coreclr/blob/master/src/coreclr/hosts/inc/coreclrhost.h
+#include "coreclrhost.h"
+
+#define MANAGED_ASSEMBLY "ManagedLibrary.dll"
+
+// Define OS-specific items like the CoreCLR library's name and path elements
+#if WINDOWS
+#include <Windows.h>
+#define FS_SEPERATOR "\\"
+#define PATH_DELIMITER ";"
+#define CORECLR_FILE_NAME "coreclr.dll"
+#elif LINUX
+#include <dirent.h>
+#include <dlfcn.h>
+#include <limits.h>
+#define FS_SEPERATOR "/"
+#define PATH_DELIMITER ":"
+#define MAX_PATH PATH_MAX
+#define CORECLR_FILE_NAME "libcoreclr.so"
+#endif
+
+// Function pointer types for the managed call and callback
+typedef int (*report_callback_ptr)(int progress);
+typedef const char* (*doWork_ptr)(const char* jobName, int iterations, int dataSize, double* data, report_callback_ptr callbackFunction);
+
+void BuildTpaList(const char* directory, const char* extension, std::string& tpaList);
+int ReportProgressCallback(int progress);
+
+int main(int argc, char* argv[])
+{
+    // Get the current executable's directory
+    // This sample assumes that both CoreCLR and the
+    // managed assembly to be loaded are next to this host
+    // so we need to get the current path in order to locate those.
+    char runtimePath[MAX_PATH];
+#if WINDOWS
+    GetFullPathNameA(argv[0], MAX_PATH, runtimePath, NULL);
+#elif LINUX
+    realpath(argv[0], runtimePath);
+#endif
+    char *last_slash = strrchr(runtimePath, FS_SEPERATOR[0]);
+    if (last_slash != NULL)
+        *last_slash = 0;
+
+    // Construct the CoreCLR path
+    // For this sample, we know CoreCLR's path. For other hosts,
+    // it may be necessary to probe for coreclr.dll/libcoreclr.so
+    std::string coreClrPath(runtimePath);
+    coreClrPath.append(FS_SEPERATOR);
+    coreClrPath.append(CORECLR_FILE_NAME);
+
+    // Construct the managed library path
+    std::string managedLibraryPath(runtimePath);
+    managedLibraryPath.append(FS_SEPERATOR);
+    managedLibraryPath.append(MANAGED_ASSEMBLY);
+    
+    //
+    // STEP 1: Load CoreCLR (coreclr.dll/libcoreclr.so)
+    //
+#if WINDOWS
+    // <Snippet1>
+    HMODULE coreClr = LoadLibraryExA(coreClrPath.c_str(), NULL, 0);
```

Comment:
@morganbr I realize this is an example, but examples are what users tend to mimic when writing their own solutions. What would you recommend we provide as an example for using the `LoadLibraryExA` API properly?